dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentsAPIImpl.java

```diff
@@ -95,9 +107,91 @@ public Experiment save(final Experiment experiment, final User user) throws
             MetricsUtil.INSTANCE.validateGoals(experiment.goals().get());
         }
 
+        if(experiment.targetingConditions().isPresent()) {
+            saveTargetingConditions(experiment, user);
+        }
+
         final Experiment experimentToSave = builder.build();
 
-        return factory.save(experimentToSave);
+        factory.save(experimentToSave);
+
+        DotPreconditions.isTrue(experimentToSave.id().isPresent(), "Experiment doesn't have Id");
+
+        final Optional<Experiment> savedExperiment = find(experimentToSave.id().get(), user);
+
+        DotPreconditions.isTrue(savedExperiment.isPresent(), "Saved Experiment not found");
+
+        return savedExperiment.get();
+    }
+
+    private void saveTargetingConditions(final Experiment experiment, final User user)
+            throws DotDataException, DotSecurityException {
+        if(experiment.targetingConditions().isEmpty()) {
+            return;
+        }
+
+        List<Rule> rules = Try.of(()->rulesAPI
+                .getAllRulesByParent(experiment, user, false))
+                .getOrElse(Collections::emptyList);
+
+        Rule experimentRule;
+
+        if(UtilMethods.isSet(rules)) {
+            experimentRule = rules.get(0);
+        } else {
+            experimentRule = createRuleAndConditionGroup(experiment, user);
+        }
+
+        // transform and save TargetingConditions into conditions
+        experiment.targetingConditions().get().forEach((targetingCondition -> {
+            createAndSaveCondition(user, experimentRule, targetingCondition);
+        }));
+
+    }
+
+    private void createAndSaveCondition(User user, Rule experimentRule,
+            TargetingCondition targetingCondition) {
+        Condition condition = targetingCondition.id().isPresent()
+            ? Try.of(()->rulesAPI.getConditionById(targetingCondition.id().get(), user, false))
```

Comment:
Probably cleaner:
```java
Condition condition = targetingCondition.id()
            .map(id -> Try.of(()->rulesAPI.getConditionById(id, user, false))
                            .getOrElseThrow(()->new IllegalArgumentException("Invalid targeting Condition Id provided. Id: " + id))).orElse(createCondition(experimentRule, targetingCondition));
```

Reply:
Gonna leave like it is so it's clear that it is based on the optional being present or not